webapp/src/ts/modules/reports/reports.component.ts

```diff
@@ -190,6 +194,14 @@ export class ReportsComponent implements OnInit, AfterViewInit, OnDestroy {
     this.subscription.add(subscription);
   }
 
+  private subscribeToResponsiveChanges() {
+    const responsiveSubscription = this.responsiveService.viewportChanged$.subscribe(() => {
+      this.getFastActionButtonType();
+      this.changeDetectorRef.detectChanges();
```

Comment:
Is there another way? We avoid using `this.changeDetectorRef.detectChanges();` because it gives performance overhead and can get stuck in change detection loops and risk undesired behaviors because it breaks Angular's detection change cycle. 

It exists for some special cases where people handle the detection change manually, which we don't

Reply:
The other solution I see is using Angular's [AsyncPipe](https://angular.dev/api/common/AsyncPipe?tab=description) in the template. We use the `responsiveService.viewportChanged$` observable then update the template when value changes.

```
<mm-fast-action-button
  [config]="{ button: { type: fastActionButtonType$ | async } }"
  [fastActions]="fastActionList">
</mm-fast-action-button>
```

Reply:
I found an alternative that reduces complexity (having fewer observables) and relies on CSS more. I just did it for reports, and if it works, then we need to do it for the Message and Contact pages

https://github.com/medic/cht-core/tree/angular-exeption-in-fab

Give it a try, and please let me know what you think

Reply:
Yes that worked and is a MUCH SIMPLER solution. We also need to add the following to `@media (max-width: @media-mobile) {}`

```
.mobile-only {
    display: initial;
  }
  .desktop-only {
    display: none;
  }
  ```

Reply:
Maybe double-check where those classes are currently used, in case they were intentionally skipped from `@media (max-width: @media-mobile) {}`

Reply:
I've checked where the classes are used, compared with master ... adding the classes in mobile does not change behavior. 

But without them, we get the wrong button 

<img width="535" alt="Screenshot 2024-08-30 at 11 50 59" src="https://github.com/user-attachments/assets/81724c18-0a6e-46aa-bba4-9c3bd62c39d9">

Reply:
Thanks for checking Ben!